### PR TITLE
Fix tickets

### PR DESCRIPTION
**Pull request by bear454**
Resolve ~two~three small issues on ticketing:

* Prevent math errors in calculating ticket 'turnover' _(yes this can still be a problem)_
* Gracefully fail out of image handling in ticket PDF ( I see this issue, in particular, when using a database copy to test, that doesn't have access to the same Cloudinary account - e.g. the same asset host).
* ticket_purchases_total no longer requires a redunant ID. `Ticket.find(X).ticket_puchases_total(X)` ?!

### CODE REVIEW

File: app/models/ticket.rb [hunk 1]
```diff
@@ -64,10 +64,13 @@ def self.total_price_user(conference, user, paid: false)
   end
 
   def tickets_turnover_total(id)
-    ticket = Ticket.find(id)
-    return Money.new(0, 'USD') unless ticket
-    sum = ticket.ticket_purchases.paid.total
-    Money.new(sum, ticket.price_currency)
+    tickets = TicketPurchase.where(ticket_id: id).paid
```
Comment (Ana06) [inline app/models/ticket.rb]: what is the difference between this code and the one before? :thinking:

File: app/models/ticket.rb [hunk 2]
```diff
@@ -64,10 +64,13 @@ def self.total_price_user(conference, user, paid: false)
...
+    if tickets.blank?
+      Money.new(0, Ticket.find(id).price_currency)
+    else
+      total = tickets.inject(0) { |sum, ticket| sum + (ticket.price_cents * ticket.quantity) }
```
Comment (Ana06) [inline app/models/ticket.rb]: `sum` instead of `inject` :see_no_evil:

File: app/models/ticket.rb [hunk 3]
```diff
@@ -64,10 +64,13 @@ def self.total_price_user(conference, user, paid: false)
   end
 
   def tickets_turnover_total(id)
-    ticket = Ticket.find(id)
-    return Money.new(0, 'USD') unless ticket
-    sum = ticket.ticket_purchases.paid.total
-    Money.new(sum, ticket.price_currency)
+    tickets = TicketPurchase.where(ticket_id: id).paid
+    if tickets.blank?
```
Comment (Ana06) [inline app/models/ticket.rb]:
```ruby
return Money.new(0, Ticket.find(id).price_currency) if tickets.blank?
```
instead of `if-else`

**Comment by bear454** [comment]
@Ana06 can you please review again?

**Comment by bear454** [comment]
Fair enough, @Ana06 ; I'll analyze the data, and see if I can find some other underlying problem. In the meantime, I split out the PDF fix  into PR#2221.

**Pull request closed without merging**